tools/retrive_go_repos.py

```diff
@@ -0,0 +1,302 @@
+        except URLError:
+            get_dependancy_data_from_vendor_folder(org_repo)
+
+
+def get_commit_sha(org_repo: str):
+    """Get commit sha for a github repo."""
+    commit_raw_url = "https://api.github.com/repos/{org_repo}/commits/master"
+    url = commit_raw_url.format(org_repo=org_repo)
+    try:
+        result = requests.get(url, headers={'Authorization': 'Bearer {token}'.format(token=GITHUB_ACCESS_TOKEN)})
+        if result.status_code == 200:
+            return result.json()['sha']
+        else:
+            logging.error("repo doesn't exist : " + org_repo)
+    except Exception as ex:
+        logging.error("Error while retriving commit details for repo : {repo}, msg : {msg}"
+                      .format(repo=org_repo, msg=str(ex)))
+    return None
+
+
+def get_vendor_folder_git_tree_url(org_repo: str, sha: str):
+    """Get vendor folder git tree url."""
+    repo_structure_raw_url = "https://api.github.com/repos/{org_repo}/git/trees/{sha}"
+    url = repo_structure_raw_url.format(org_repo=org_repo, sha=sha)
+
+    try:
+        result = requests.get(url, headers={'Authorization': 'Bearer {token}'.format(token=GITHUB_ACCESS_TOKEN)})
+        if result.status_code == 200:
+            vendor_folder_info = list(filter(lambda x: (x['path'] == "vendor"), result.json()['tree']))
+            if len(vendor_folder_info) > 0:
+                return vendor_folder_info[0]['url']
+        else:
+            logging.error("repo doesn't exist : " + org_repo)
+    except Exception as ex:
+        logging.error("Error while retriving repo structure details {repo}, msg: {msg}"
+                      .format(repo=org_repo, msg=str(ex)))
+
+    return None
+
+
+def get_dependancy_repo_from_vendor(git_tree_url: str, level: int, path: str):
+    """Get dependancy repo list from vendor folder."""
+    result = requests.get(git_tree_url, headers={'Authorization': 'Bearer {token}'.format(token=GITHUB_ACCESS_TOKEN)})
+    if result.status_code == 200:
+        json_data = result.json()
+        if 'tree' in json_data:
+            if level == 0:
+                for item in result.json()['tree']:
+                    get_dependancy_repo_from_vendor(item['url'], 1, item['path'])
+            elif level == 1 or level == 2:
+                for item in json_data['tree']:
+                    pkg = path + "/" + item['path']
+                    if pkg not in pkg_list:
+                        pkg_list.append(pkg)
+
+                        data = get_go_pkg_data(pkg)
+                        if 'Repository: <a href="https://github.com/' in data.text:
+                            get_dependancy_repo_url_from_text(data.text)
+                        elif '404 Not Found' in data.text and level == 1:
+                            get_dependancy_repo_from_vendor(item['url'], 2, pkg)
+
+
+def save_data_into_file():
+    """Save data into different json files."""
+    distinct_repo_urls = list(dict.fromkeys(repo_urls))
```

Comment:
Might be able to use `OrderedDict` here, please check once.

Reply:
`OrderedDict` works for only dictionary, for list uing `sorted(set(list))`

Reply:
The `dict.keys()` method of an ordered dict would give you a sorted list only, but this is ok too.